File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/HighScores.java
package exercism;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class HighScores {

    private final List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = highScores;
    }

    List<Integer> scores() {
        return highScores;
    }

    Integer latest() {
        return highScores.get(highScores.size() - 1);
    }

    Integer personalBest() {
        return Collections.max(highScores);
    }

    List<Integer> personalTopThree() {
        List<Integer> sortedScores = new ArrayList<>(highScores);
        sortedScores.sort(Collections.reverseOrder());
        return sortedScores.subList(0, Math.min(3, sortedScores.size()));
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/HighScores.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp;

using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private readonly List<int> _scoresList;

    public HighScores(List<int> list)
    {
        _scoresList = list;
    }

    public List<int> Scores()
    {
        return _scoresList;
    }

    public int Latest()
    {
        return _scoresList.Last();
    }

    public int PersonalBest()
    {
        return _scoresList.Max();
    }

    public List<int> PersonalTopThree()
    {
        return _scoresList.OrderByDescending(score => score).Take(3).ToList();
    }
}
